### ISSUE TITLE
S4 not initialised

### ISSUE THREAD

**Issue by BriJackson** [nicoh88/homebridge-xiaomi-roborock-vacuum#87]
Trying to add my S4 Roborock and receiving the messages in logs:
1/18/2020, 7:33:57 PM] [Xiaomi Mi Robot Vaccum 1st Generation] ERR getCleaning | No vacuum cleaner is discovered yet.
[1/18/2020, 7:33:57 PM] [Xiaomi Mi Robot Vaccum 1st Generation] INF ensureDevice | undefined | The socket was destroyed or not initialised, initialising the device
[1/18/2020, 7:33:57 PM] [Xiaomi Mi Robot Vaccum 1st Generation] ERR getSpeed | No vacuum cleaner is discovered yet.
[1/18/2020, 7:33:57 PM] [Xiaomi Mi Robot Vaccum 1st Generation] INF ensureDevice | undefined | The socket was destroyed or not initialised, initialising the device
[1/18/2020, 7:33:57 PM] [Xiaomi Mi Robot Vaccum 1st Generation] ERR getDocked | No vacuum cleaner is discovered yet.
[1/18/2020, 7:33:57 PM] [Xiaomi Mi Robot Vaccum 1st Generation] INF ensureDevice | undefined | The socket was destroyed or not initialised, initialising the device
[1/18/2020, 7:33:57 PM] [Xiaomi Mi Robot Vaccum 1st Generation] ERR getBatteryLow | No vacuum cleaner is discovered yet.
[1/18/2020, 7:33:57 PM] [Xiaomi Mi Robot Vaccum 1st Generation] INF ensureDevice | undefined | The socket was destroyed or not initialised, initialising the device
[1/18/2020, 7:33:57 PM] [Xiaomi Mi Robot Vaccum 1st Generation] ERR getDevice | Is not a vacuum cleaner!
[1/18/2020, 7:33:57 PM] [Xiaomi Mi Robot Vaccum 1st Generation] ERR getCleaning | Failed getting the cleaning status. TypeError: Cannot read property 'property' of null
    at XiaomiRoborockVacuum.get isCleaning [as isCleaning] (/usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/index.js:485:32)
    at XiaomiRoborockVacuum.getCleaning (/usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/index.js:493:31)
    at async module.exports (/usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/lib/callbackify.js:5:20)

**Comment by BriJackson** [comment]
Model is roborock.vacuum.s4

**Comment by nicoh88** [comment]
Please update to version `0.6.5`, S4 should be recognized.

Version 0.6.5 published:

- Feature: Add support for Roborock S5 Max
- Improve: Refresh the state every 30s to ensure miio is still properly connected
- Bugfix: Battery level
- Bugfix: UnhandledPromiseRejectionWarning on startup when it fails to connect

Install:

```
sudo su -
npm -g install homebridge-xiaomi-roborock-vacuum@latest
```

**Comment by edelmaca** [comment]
i got the 
"INF ensureDevice | undefined | The socket was destroyed or not initialised, initialising the device"
message too.
If i extract the token from the MiHome App and use the Roborock App afterwards, the token is resetting every time and i get this message. After this, my whole Homebridge Setup ist completely dead. Is it somehow possible to get it to work with the Roborock App?

**Comment by nicoh88** [comment]
**Successfully published homebridge-xiaomi-roborock-vacuum@0.6.7**

```
sudo su -
npm -g install homebridge-xiaomi-roborock-vacuum@latest
```
